src/learner.cc

```diff
@@ -443,9 +443,12 @@ class LearnerImpl : public Learner {
 
   void Predict(DMatrix* data, bool output_margin,
                std::vector<bst_float>* out_preds, unsigned ntree_limit,
-               bool pred_leaf, bool pred_contribs, bool approx_contribs) const override {
-    if (pred_contribs) {
+               bool pred_leaf, bool pred_contribs, bool approx_contribs,
+               bool interaction_contribs) const override {
+    if (pred_contribs && !interaction_contribs) {
       gbm_->PredictContribution(data, out_preds, ntree_limit, approx_contribs);
+    } else if (pred_contribs && interaction_contribs) {
```

Comment:
Is it really necessary for interactions to have both pred_contribs and interaction_contribs on?

Reply:
Good point. I'll just let interaction_contribs stand alone